clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -962,9 +964,36 @@ CIRGenModule::getConstantArrayFromStringLiteral(const StringLiteral *e) {
     return builder.getString(str, eltTy, finalSize);
   }
 
-  errorNYI(e->getSourceRange(),
-           "getConstantArrayFromStringLiteral: wide characters");
-  return mlir::Attribute();
+  auto arrayTy = mlir::cast<cir::ArrayType>(convertType(e->getType()));
+
+  auto arrayEltTy = mlir::cast<cir::IntType>(arrayTy.getElementType());
+
+  uint64_t arraySize = arrayTy.getSize();
+  unsigned literalSize = e->getLength();
+  assert(arraySize == literalSize &&
```

Comment:
I see that this assert is firing in testing. I wasn't expecting that. Looking closer at the classic codegen implementation, it fills the `elements` array from zero to `E->getLength()` (i.e. `literalSize` here) and then resizes the array to `NumElements`. Given that classic codegen is using a vector of integers, maybe the resize handles the zero padding. I guess you need to bring back the zero-padding loop. Sorry about the bad suggestion.